core/primitives/src/receipt.rs

```diff
@@ -113,6 +115,24 @@ pub struct StateStoredReceiptV0<'a> {
     pub metadata: StateStoredReceiptMetadata,
 }
 
+/// The V1 of StateStoredReceipt.
+#[derive(BorshDeserialize, BorshSerialize, PartialEq, Eq, Debug, ProtocolSchema)]
+pub struct StateStoredReceiptV1<'a> {
+    pub receipt: Cow<'a, Receipt>,
```

Comment:
@wacban, maybe a good time to cleanup the Cow part of the receipt here which you've been meaning to do.